<?xml version="1.0"?>
<launch>
  <!-- rviz -->
    <node pkg="rviz" type="rviz" name="kitti_rviz" required="true" 
      args = "-d $(find fusion_nodelet)/rviz/kitti.rviz" />
  <!-- urdf -->
    <!-- <arg name="gui" default="False" />
    <param name="robot_description" textfile="$(find fusion_nodelet)/urdf/fusion.urdf" />
    <param name="use_gui" value="$(arg gui)"/>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" /> -->
  <!-- rosbag play -->
    <node pkg="rosbag" type="play" name="fusion_bag_play" args="-l -r 1.0 $(env HOME)/data/bag/kitti_2011_09_26_drive_0057_synced.bag" />
  
  <!-- nodelet*************nodelet -->

  <!-- nodelet::Global Arg -->
    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find fusion_nodelet)/config/lidar_perception.config"/>
    <arg name="arg_nodelet_manager" default="velodyne_nodelet_manager" />
    <arg name="arg_pose_path" default="$(find fusion_nodelet)/config/calib_lidar.txt"/>

  <!-- nodelet -->
    <node pkg="nodelet" type="nodelet" name="$(arg arg_nodelet_manager)" args="manager" output="screen"/>
  
  <!-- nodelet::PerceptionNodelet -->
    <node pkg="nodelet" type="nodelet" name="perception_nodelet" args="load fusion_nodelet/PerceptionNodelet $(arg arg_nodelet_manager)">
      <param name="perception_cloud_topic" value="/kitti/velo/pointcloud"/>
      <param name="h_lidar" value="2.5"/>
    </node>
</launch>
